content/en/docs/languages/php/context.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Context
+weight: 55
+cSpell:ignore: Swoole
+---
+
+In order for OpenTelemetry to work, it must store and propagate important
+telemetry data. For example, when a request is received and a span is started it
+must be available to a component which creates its child span. To solve this
+problem, OpenTelemetry stores the span in the Context.
+
+More information:
+
+- [Context specification](/docs/specs/otel/context/)
+
+## Context
+
+Context is globally available, and there can only be one
+[active context](#active-context) in the current execution context. Context can
+store values (for example, a `Span`).
+
+### Storage
+
+Context uses `Storage` to keep track of values. By default, a generic
+`ContextStorage` is used. OpenTelemetry for PHP supports other context storage
+for more exotic use-cases (eg for asynchronous/concurrent execution with
+`fibers`).
+
+## Context Keys
+
+Context entries are key-value pairs. Keys can be created by calling
+`OpenTelemetry\Context\Context::createKey()`.
+
+```php
+use OpenTelemetry\Context\Context;
+
+$key1 = Context::createKey('My first key');
+$key2 = Context::createKey('My second key');
+```
+
+Context keys are used to store and retrieve values from context.
+
+## Basic Operations
+
+### Store and Retrieve values
+
+Values are stored in Context by using the `$context->with($key, $value)` method.
+Setting a context entry creates a new context with the new entry in its storage,
+containing `$value`.
+
+Context is immutable, and setting a context entry does not modify the previous
+context. Once a value is stored in a context, it can be retrieved via
+`$context->get($key)`:
+
+```php
+use OpenTelemetry\Context\Context;
+
+$key = Context::createKey('some key');
+
+// add a new entry
+$ctx2 = Context::getCurrent()->with($key, 'context 2');
+
+// ctx2 contains the new entry
+var_dump($ctx2->get($key)); // "context 2"
+
+// active context is unchanged
+var_dump(Context::getCurrent()->get($key)); // NULL
+```
+
+If a value is not found in the current context, then each parent is checked
+until either the key is found, or the root context is reached.
+
+## Active Context
+
+The active context is the context which is returned by `Context::getCurrent()`.
+The context object contains entries which allow telemetry components to
+communicate with each other. For example, when a span is created it may be
+activated, which creates a new active context and stores the span. Later, when
+another span is created it may use the span from the active context as its
+parent span. If no context is active, the root context is returned, which is
+just the empty context object.
```

Comment:
```suggestion
communicate with each other. For example, when a span is created it can be
activated, which creates a new active context and stores the span. Later, when
another span is created it can use the span from the active context as its
parent span. If no context is active, the root context is returned, which is
just the empty context object.
```